node/actors/bft/src/tests/unit_tests.rs

```diff
@@ -0,0 +1,599 @@
+    );
+}
+
+// #[tokio::test]
+// async fn replica_commit_sanity() {
+//     let mut util = Util::make().await;
+//
+//     let replica_prepare = util::make_replica_prepare(&consensus, None::<fn(&mut ReplicaPrepare)>);
+//     let res = util::dispatch_replica_prepare(&ctx, &mut consensus, replica_prepare);
+//     assert_eq!(res, Ok(()));
+//     let leader_prepare = util::make_leader_prepare_from_replica_prepare(&consensus, &mut rng, replica_prepare, Some(|msg: &mut LeaderPrepare| {
+//         msg.view = ViewNumber(2);
+//     }));
+//     let replica_commit = util::make_replica_commit(&consensus, leader_prepare.msg., None::<fn(&mut ReplicaCommit)>);
+//
+//     let res = scope::run!(&ctx, |ctx, s| {
+//         s.spawn_blocking(|| {
+//             let res = consensus
+//             .leader
+//             .process_replica_commit(ctx, &consensus.inner, leader_prepare.cast().unwrap());
+//             Ok(res)
+//         })
+//         .join(ctx)
+//     })
+//         .await
+//         .unwrap();
+//
+//     assert_eq!(
+//         res,
+//         Err(LeaderPrepareError::InvalidPrepareQC(anyhow!("PrepareQC contains messages for different views!")))
+//     );
+// }
+
+mod util {
+    use crate::{
+        io::{InputMessage, OutputMessage},
+        leader::ReplicaPrepareError,
+        replica::LeaderPrepareError,
+        Consensus,
+    };
+    use rand::{rngs::StdRng, Rng, SeedableRng};
+    use zksync_concurrency::{ctx, ctx::Ctx, scope};
+    use zksync_consensus_network::io::ConsensusInputMessage;
+    use zksync_consensus_roles::{
+        validator,
+        validator::{
+            BlockHeader, ConsensusMsg, LeaderPrepare, Payload, Phase, PrepareQC, ReplicaCommit,
+            ReplicaPrepare, SecretKey, Signed, ViewNumber,
+        },
+    };
+    use zksync_consensus_utils::pipe::DispatcherPipe;
+
+    pub(crate) struct Util {
+        ctx: Ctx,
+        consensus: Consensus,
+        rng: StdRng,
+        pipe: DispatcherPipe<InputMessage, OutputMessage>,
+        keys: Vec<SecretKey>,
+    }
+
+    impl Util {
+        pub async fn make() -> Util {
+            Util::make_with(1).await
+        }
+
+        pub async fn make_with(num_validators: i32) -> Util {
+            let ctx = ctx::test_root(&ctx::RealClock);
+            let mut rng = StdRng::seed_from_u64(6516565651);
+            let keys: Vec<_> = (0..num_validators).map(|_| rng.gen()).collect();
+            let (genesis, val_set) =
+                crate::testonly::make_genesis(&keys, validator::Payload(vec![]));
+            let (mut consensus, pipe) =
+                crate::testonly::make_consensus(&ctx, &keys[0], &val_set, &genesis).await;
+
+            consensus.leader.view = ViewNumber(1);
+            consensus.replica.view = ViewNumber(1);
+
+            Util {
+                ctx,
+                consensus,
+                rng,
+                pipe,
+                keys,
+            }
+        }
+
+        pub fn own_key(&mut self) -> &SecretKey {
```

Comment:
You don't need `&mut self` here and in `key_at`; `&self` would suffice.

Reply:
Fixed.